File: programmers/python/programmers_sort_2.py
def solution(numbers):
	answer = ''
	numbers = [str(x) for x in numbers]
	numbers.sort(key=lambda x: (x * 4)[:4], reverse=True)
	if numbers[0] == '0':
		answer = '0'
	else:
		for x in numbers:
			answer += x
	return answer

File: programmers/cpp/programmers_sort_2.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(const string &a, const string &b)
{
	string c = a + b;
	string d = b + a;

	if (atoi(c.c_str()) > atoi(d.c_str()))
		return true;
	else
		return false;
}

string solution(vector<int> numbers) {
	string answer = "";
	vector<string> v;

	for (int i = 0; i < numbers.size(); i++)
		v.push_back(to_string(numbers[i]));
	sort(v.begin(), v.end(), cmp);
	for (int i = 0; i < v.size(); i++)
		answer += v[i];

	if (answer[0] == '0')
		return "0";

	return answer;
}